py/server/deephaven/column.py

```diff
@@ -53,7 +58,7 @@ def name(self) -> str:
     @cached_property
     def data_type(self) -> DType:
         """The column data type."""
-        return from_jtype(self.j_column_definition.getDataType())
+        return from_jtype(self.j_column_definition.getDataType())  # type: ignore[return-value]
```

Comment:
Code smell.  I think this is correct and is calling java, but give it a once over.